### ISSUE TITLE
[P1][dogfood] Scala 3 `enum` declarations entirely dropped from extraction (silent success, class_count=0, no enum_definition handler in scala_plugin)

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#762]
**Severity:** P1  |  **Surface:** Scala extraction (--advanced --table json) — tree_sitter_analyzer/languages/scala_plugin.py
_Found by parallel dogfood audit round 2 on develop @ 471ac5f5; adversarially re-reproduced + dup-checked by an independent verifier._

### Observed
classes: [] (empty); statistics.class_count: 0; success: true; verdict: INFO; agent_summary.summary_line 'classes=0 methods=1 fields=0 imports=0 lines=11'. Both `enum Color` and `enum Planet` are missing entirely; only the method `hex` survives, orphaned (no parent class — methods[] has one entry 'hex' with is_method/parent context absent). The brace-less Scala 3 form `enum Color:` is also dropped (classes=[], class_count=0, success=True). No warning emitted; the verdict reports a clean successful extraction. The same plugin correctly extracts class/trait/object (Foo->class, Bar->trait, Baz->object). Grammar confirmed sane: tree-sitter-scala parses `enum Color` as an `enum_definition` node; scala_plugin.py:167-169 dispatch has no enum_definition handler, so the node is silently skipped.

### Expected
Each `enum` should be extracted as a class-like entry with type 'enum' (the sibling PHP plugin already does exactly this: `enum UserStatus: string` -> {name:'UserStatus', type:'enum'}, class_count:1). For the repro file, class_count should be 2 (Color, Planet) with their methods/cases attached to the correct parent, and `hex` should not be orphaned with parent_class=None. The agent_summary should not imply the file has no types. Both braced and brace-less (significant-indentation) Scala 3 enum forms must be captured. Dropping a primary Scala 3 ADT construct silently (success:true, no warning) is a correctness failure an agent would never detect.

### Repro
```bash
cd /Users/aisheng.yu/git-private/tsa-dogfood (develop)
mkdir -p _p
printf 'package demo\n\nenum Color {\n  case Red, Green, Blue\n  def hex: String = "x"\n}\n\nenum Planet(mass: Double) {\n  case Earth extends Planet(5.97)\n  case Mars extends Planet(6.42)\n}\n' > _p/ScalaEnum.scala
uv run python -m tree_sitter_analyzer _p/ScalaEnum.scala --advanced --table json

Brace-less form also fails:
printf 'enum Color:\n  case Red, Green, Blue\n' > _p/ScalaEnum2.scala
uv run python -m tree_sitter_analyzer _p/ScalaEnum2.scala --advanced --table json

Sibling PHP plugin DOES extract enum (contrast):
printf '<?php\nenum UserStatus: string {\n  case Active = "active";\n}\n' > _p/PhpEnum.php
uv run python -m tree_sitter_analyzer _p/PhpEnum.php --advanced --table json
```

### Verifier notes
Reproduced the exact repro on tsa-dogfood/develop: `enum Color` and `enum Planet` are both entirely absent from the extraction output — classes: [] (empty), statistics.class_count: 0, success: true, verdict: INFO, agent_summary.summary_line 'classes=0 methods=1 fields=0 imports=0 lines=11'. The only surviving symbol is the orphaned method `hex` (parent_class=None). The brace-less Scala 3 form `enum Color:` is also dropped (verified separately: classes=[], class_count=0, success=True). Root cause located in tree_sitter_analyzer/languages/scala_plugin.py lines 167-169: the node-type dispatch handles class_definition/object_definition/trait_definition but has NO enum_definition handler, so the node is silently skipped. This is NOT a parse failure: I parsed `enum Color { ... }` with tree-sitter-scala directly and it produces a clean `enum_definition` node — the grammar supports it; the plugin just ignores it. NOT a flag/user error: --advanced --table json is the documented extraction path and correctly extracts class('Foo','class'), trait('Bar','trait'), object('Baz','object') from the same plugin. PHP-parity claim confirmed: PHP plugin extracts `enum UserStatus: string` as {name:'UserStatus', type:'enum', class_count:1}; Scala does not. Dedup: searched open issues incl. the full #734-757 round-1 batch — that batch covers JS/Go/C/C++ extraction bugs but no Scala enum. #753 (C anonymous nested aggregates) is a different language and a different mechanism (anon containers vs. an entirely unhandled top-level node type), not a dup. No matching open issue found.

**Comment by aimasteracc** [comment]
Fixed: scala_plugin.py lines 395-404 now handle enum_definition nodes, extracting Color/Planet as class_type='enum'. Verified: class_count=7 (2 enums + 5 enum members) on current develop.
